libpd_wrapper/x_libpdreceive.c

```diff
@@ -48,24 +48,36 @@ static void libpdrecanything(t_libpdrec *x, t_symbol *s,
 }
 
 static void libpdreceive_free(t_libpdrec *x) {
-    pd_unbind(&x->x_obj.ob_pd, x->x_sym);
+  pd_unbind(&x->x_obj.ob_pd, x->x_sym);
 }
 
-void *libpdreceive_new(t_symbol *s) {
-  t_libpdrec *x = (t_libpdrec *)pd_new(libpdrec_class);
+static void *libpdreceive_donew(t_symbol *s) {
+  t_libpdrec *x;
+  x = (t_libpdrec *)pd_new(libpdrec_class);
   x->x_sym = s;
   pd_bind(&x->x_obj.ob_pd, s);
   return x;
 }
 
+    /* This is exposed in the libpd API so must set the lock. */
+void *libpdreceive_new(t_symbol *s) {
+  t_libpdrec *x;
+  sys_lock();
```

Comment:
We don't need the lock/unlock pair here as libpdreceive_donew() locks internally.